### PR TITLE
Update Realm Objective-C's Swift Tests for Swift 3`

### PR DESCRIPTION
**Pull request by JadenGeller**
We currently don't have *any* tests for Realm Objective-C in Swift 3.0, so I updated the Swift 2.x ones. https://github.com/realm/realm-cocoa/issues/3842

### CODE REVIEW

**Comment by JadenGeller** [comment]
@austinzheng @bdash

File: Realm/RLMProperty.mm [hunk 1]
```diff
@@ -342,6 +342,16 @@ - (instancetype)initSwiftPropertyWithName:(NSString *)name
 
     id propertyValue = [obj valueForKey:_name];
 
+    // temporarily workaround since Objective-C generics used in Swift show up as `@`
```
Comment (bdash) [inline Realm/RLMProperty.mm]: Please make this a `FIXME: `, add information about which versions of Xcode and Swift this problem occurs in, and add a link to the Radar and/or swift.org bug report. It's important to make it easy for a future reader to determine whether this block of code can go away.

**Comment by jpsim** [comment]
I agree with @bdash's comment on adding more info to your objc comment. Otherwise 👍. I didn't do a thorough review of the Swift 3 tests, but we'll need to do that again when Xcode 8 goes GM anyway.

**Comment by JadenGeller** [comment]
Addresses https://github.com/realm/realm-cocoa/issues/3843

**Pull request merged**